leetcodePython/PrefixSum/prefixsum_2483.py

```diff
@@ -0,0 +1,12 @@
+class Solution:
+    def bestClosingTime(self, customers: str) -> int:
+        earliest = penalty = min_penalty = 0
+        for close, c in enumerate(customers):
+            if c == "Y":
+                penalty -= 1
+            else:
+                penalty += 1
```

Comment:
1. we start from beginning. So for example we are closing at `i`.
2. We only care after closing, how many customer come after close. Then we might have less penalty.
3. At the same time if no customer, it also means we have more penalty. 
4. If less penalty, that means we should close later.